### PR TITLE
Deletion concept implementation

### LINKED ISSUE #26: content update and deletion
# Description
When chapters are deleted, all content related to this chapter has to be deleted. Also, the content service needs to notify other services of its deletions, as some other services are dependent on content,
In the case of moving content to a different chapter, dependent services need to be notified of these changes. Dependant services are all resource services (such as Media- or Flashcard service) and the course service (Course-Resource-Associations).

# Parent 
https://github.com/IT-REX-Platform/course_service/issues/29

# Acceptance criteria
- [ ] content service receives deletion message from course service and deletes corresponding content
- [ ] content service informs dependant services of content deletion
- [ ] content service informs dependant services of update to content

### PR DESCRIPTION
**Pull request by BlueWaves2**
## Description of changes
implements functions to conform to deletion concept. 
For this content dependant Services are informed over deletion of Content Entities.
Additionally in the case of a content being linked to a different chapter, all dependant Services are informed.
This allows content dependant services to start the update mechanism to update Course-Resource-Associations (in course service).

Closes #26 
  
## How has this been tested?

- [X] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test

## Checklist before requesting a review
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My code fulfills all acceptance criteria
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation
- [X] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

**Review by PaulBredl — APPROVED** [review]

**Review by PaulBredl — APPROVED** [review]

File: src/test/java/de/unistuttgart/iste/gits/content_service/service/ContentServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ContextConfiguration(classes = MockTopicPublisherConfiguration.class)
+class ContentServiceTest {
```
[the remark this answers is not in the document]
Comment (BlueWaves2) [reply src/test/java/de/unistuttgart/iste/gits/content_service/service/ContentServiceTest.java:30]: I did this in this class to preserve consistency with other services, where those are also added to the existing test class for each service. Note, these methods are also located in the service
